ServerCore/Areas/Identity/Pages/Account/Edit.cshtml.cs

```diff
@@ -38,6 +38,15 @@ public async Task<IActionResult> OnGetAsync(int userId, string returnUrl = null)
 
         public async Task<IActionResult> OnPostAsync(string returnUrl = null)
         {
+            if (string.IsNullOrEmpty(PuzzleUser.Email))
```

Comment:
Would it work to at [Required] to PuzzleUser.Email and Team.PrimaryContactEmail? That would do this automatically